### PR TITLE
fix: update bedrock guardrail cfn primary id to be arn

### PR DESCRIPTION
**Pull request by yiyuan-he**
### *Summary of Changes:*

- Updating the CFN Primary Id for Guardrail to be the ARN.  (Verified this in the AWS Cloudformation Console)

  Screenshot 2024-10-09 at 12 44 12 PM

### *Testing Plan:*

Set up sample apps and manually verified correct span attributes were generated via auto-instrumentation. 

bedrock-guardrail-span-data-verification-python

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-opentelemetry-distro/src/amazon/opentelemetry/distro/_aws_metric_attribute_generator.py [hunk 1]
```diff
@@ -423,6 +424,7 @@ def _set_remote_type_and_identifier(span: ReadableSpan, attributes: BoundedAttri
         elif is_key_present(span, AWS_BEDROCK_GUARDRAIL_ID):
             remote_resource_type = _NORMALIZED_BEDROCK_SERVICE_NAME + "::Guardrail"
             remote_resource_identifier = _escape_delimiters(span.attributes.get(AWS_BEDROCK_GUARDRAIL_ID))
```
Comment (zzhlogin) [inline aws-opentelemetry-distro/src/amazon/opentelemetry/distro/_aws_metric_attribute_generator.py:426]: Same as java, we want to "extract remote_resource_identifier from cloudformation_primary_identifier to make sure the consistency of two values."? So the `AWS_BEDROCK_GUARDRAIL_ID` won't be directly used here?

**Review by zzhlogin — DISMISSED** [review]

File: aws-opentelemetry-distro/src/amazon/opentelemetry/distro/_aws_metric_attribute_generator.py (hunk 1 above)
Comment (yiyuan-he) [reply aws-opentelemetry-distro/src/amazon/opentelemetry/distro/_aws_metric_attribute_generator.py:426]: Left comment in that [PR's thread](https://github.com/aws-observability/aws-otel-java-instrumentation/pull/905/files#r1796076219).

File: aws-opentelemetry-distro/src/amazon/opentelemetry/distro/patches/_bedrock_patches.py [hunk 2]
```diff
@@ -217,6 +219,13 @@ def on_success(self, span: Span, result: _BotoResultT):
                 guardrail_id,
             )
 
+        guardrail_arn = result.get(_GUARDRAIL_ARN)
+        if guardrail_arn:
+            span.set_attribute(
+                AWS_BEDROCK_GUARDRAIL_ARN,
+                guardrail_arn,
+            )
```
Comment (mxiamxia) [inline aws-opentelemetry-distro/src/amazon/opentelemetry/distro/patches/_bedrock_patches.py:227]: same to https://github.com/aws-observability/aws-otel-java-instrumentation/pull/905/files#r1796412547

Comment (yiyuan-he) [reply aws-opentelemetry-distro/src/amazon/opentelemetry/distro/patches/_bedrock_patches.py:227]: Sounds good, [will go with you and Zhonghao's recommendation](https://github.com/aws-observability/aws-otel-java-instrumentation/pull/905#discussion_r1797187133).

**Review by zzhlogin — APPROVED** [review]

**Review by mxiamxia — APPROVED** [review]

**Pull request merged**